### PR TITLE
Fix x86_64

### PR DESCRIPTION
**Pull request by jamcleod**
- Add bzImage building (and enable for x86_64)
- Fix clobbering issue in hypercall implementation
- Add other various config options

### CODE REVIEW

File: configs/4.10/x86_64 [hunk 1]
```diff
@@ -1,3 +1,21 @@
 #include "x86-common.inc"
 
 CONFIG_SMP=y
+CONFIG_COMPAT=y
+CONFIG_SERIAL_NONSTANDARD=y
```
Comment (AndrewFasano) [inline configs/4.10/x86_64:5]: Why nonstandard?

File: configs/4.10/x86_64 [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+CONFIG_SERIAL_8250=y
+CONFIG_SERIAL_8250_CONSOLE=y
+CONFIG_SERIAL_8250_NR_UARTS=32
+CONFIG_SERIAL_8250_EXTENDED=y
+CONFIG_SERIAL_8250_MANY_PORTS=y
+CONFIG_SERIAL_8250_SHARE_IRQ=y
+CONFIG_SERIAL_8250_DETECT_IRQ=y
+CONFIG_SERIAL_8250_RSA=y
```
Comment (AndrewFasano) [inline configs/4.10/x86_64:13]: If these are unnecessary, could we delete? If they're required that's fine.

File: configs/4.10/x86_64 [hunk 3]
```diff
@@ -1,3 +1,21 @@
...
+
+CONFIG_IA32_EMULATION=y
+CONFIG_VIRTIO_RING=y
+
+CONFIG_VIRTIO_FS=y
+CONFIG_PCI_HOST_GENERIC=y
+CONFIG_PCI_QUIRKS=y
+CONFIG_BLK_DEV_SD=y
```
Comment (AndrewFasano) [inline configs/4.10/x86_64:21]: Same point for these being necessary or not.

**Review by AndrewFasano — APPROVED** [review]
A few small questions, but this looks good.

**Comment by jamcleod** [comment]
Table of combinations of options I've tried for serial:

|                                                  | With `SERIAL_8250` options | Without |
|-|-|-|
| With `SERIAL_NONSTANDARD` | ✅ | :x: |
| Without                                     | :x: | :x: |

`SERIAL_NONSTANDARD` is definitely necessary. Some combination of the 8250 options are also necessary.

`CONFIG_PCI_QUIRKS` and `CONFIG_BLK_DEV_SD` do not seem necessary to boot so can possibly be removed in the future.

**Pull request merged**